### PR TITLE
Money dupe patch

### PR DESCRIPTION
**Pull request by DESPEL**
Patching simple money dupe

Steps to reproduce the dupe

1. Trade and put all your money
image
image
2. Lose money somehow, I.E. eco take (user) (amount)
image
3.Complete the trade
 The money is not substracted
image
The money is added to the other account
image

The changes just add a verification before completing the trade.

### CODE REVIEW

**Comment by DESPEL** [comment]
Solves https://github.com/CodingAir/TradeSystem/issues/209

File: src/main/java/de/codingair/tradesystem/trade/Trade.java [hunk 1]
```diff
@@ -232,6 +232,14 @@ private void finish() {
         Runnable runnable = new Runnable() {
             @Override
             public void run() {
+                // code to avoid some weird money dupe
+                Profile p0 = TradeSystem.getProfile(players[0]);
+                Profile p1 = TradeSystem.getProfile(players[1]);
+                if (p0.getMoney() > money[0] || p1.getMoney() > money[1]) {
+                    cancel(Lang.getPrefix() + Lang.get("Economy_Error"));
+                    return;
```
Comment (CodingAir) [inline src/main/java/de/codingair/tradesystem/trade/Trade.java:240]: Leaves the GUI open.

File: src/main/java/de/codingair/tradesystem/trade/Trade.java [hunk 2]
```diff
@@ -287,9 +295,6 @@ public void run() {
                 guis[0].close();
                 guis[1].close();
 
-                Profile p0 = TradeSystem.getProfile(players[0]);
-                Profile p1 = TradeSystem.getProfile(players[1]);
-
```
Comment (CodingAir) [inline src/main/java/de/codingair/tradesystem/trade/Trade.java:292]: Why do you remove the Profile variables?

**Review by CodingAir — CHANGES_REQUESTED** [review]

File: src/main/java/de/codingair/tradesystem/trade/Trade.java (hunk 2 above)
Comment (DESPEL) [reply src/main/java/de/codingair/tradesystem/trade/Trade.java:292]: I used them at the money check, so I just moved them upwards.

File: src/main/java/de/codingair/tradesystem/trade/Trade.java (hunk 1 above)
Comment (DESPEL) [reply src/main/java/de/codingair/tradesystem/trade/Trade.java:240]: The GUI is closed by the cancel method (line 239)
image

File: src/main/java/de/codingair/tradesystem/trade/Trade.java (hunk 2 above)
Comment (CodingAir) [reply src/main/java/de/codingair/tradesystem/trade/Trade.java:292]: Oh yeah, sorry

File: src/main/java/de/codingair/tradesystem/trade/Trade.java (hunk 1 above)
Comment (CodingAir) [reply src/main/java/de/codingair/tradesystem/trade/Trade.java:240]: Haha, I'm blind :D Sorry. 
BUT one thing left. `p0.getMoney() > money[0]` is wrong. This triggers an Economy_Error if the player has more money  than his offer.

**Review by CodingAir — CHANGES_REQUESTED** [review]

File: src/main/java/de/codingair/tradesystem/trade/Trade.java (hunk 1 above)
Comment (DESPEL) [reply src/main/java/de/codingair/tradesystem/trade/Trade.java:240]: Fixed

**Comment by CodingAir** [comment]
Thank you for your contribution.

**Pull request merged**